Firestore/core/src/firebase/firestore/core/sync_engine.cc

```diff
@@ -429,9 +430,10 @@ void SyncEngine::EmitNewSnapshotsAndNotifyLocalStore(
       // The query has a limit and some docs were removed/updated, so we need to
       // re-run the query against the local store to make sure we didn't lose
       // any good docs that had been past the limit.
-      DocumentMap docs = local_store_->ExecuteQuery(query_view->query());
-      view_doc_changes =
-          view.ComputeDocumentChanges(docs.underlying_map(), view_doc_changes);
+      QueryResult query_result = local_store_->ExecuteQuery(
```

Comment:
Port of https://github.com/firebase/firebase-android-sdk/blob/d8ff6a54d12fd676a31c2c331f09fe0a7d033202/firebase-firestore/src/main/java/com/google/firebase/firestore/core/SyncEngine.java#L530